ibis/expr/operations/relations.py

```diff
@@ -126,6 +126,16 @@ def to_pyarrow(self, schema: Schema) -> pa.Table:
         return self._data
 
 
+class PyArrowRBRProxy(TableProxy):
+    __slots__ = ()
+
+    def to_frame(self):
+        return self._data.read_pandas()
+
+    def to_pyarrow(self, schema: Schema) -> pa.Table:
+        return self._data.read_all()
```

Comment:
The `to_pyarrow` call for a `memtable` registered for a `RecordBatchReader` will only work once, meaning that the `memtable` can only ever have `execute` executed against it once. This seems like a source of confusion for users.

Reply:
Hm, that's a good point. Perhaps we need some stronger conceptual and documented requirements for something to be `memtable`-able, such as "must be replayable/repeatable/able to be consumed more than once"